parsec/scheduling.c

```diff
@@ -532,6 +531,108 @@ int __parsec_task_progress( parsec_execution_stream_t* es,
     return rc;
 }
 
+static int __parsec_taskpool_test( parsec_taskpool_t* tp, parsec_execution_stream_t *es )
+{
+    parsec_context_t* parsec_context = es->virtual_process->parsec_context;
+    parsec_task_t* task;
+    int nbiterations = 0, distance, rc;
+
+    assert(PARSEC_THREAD_IS_MASTER(es));
```

Comment:
don't think that assert is correct, what about recursive task pools
(same comment appears twice)

Reply:
We could allow usage of this function in other threads than master. That being said I'm not sure all accessors used in this function are thread safe.

Reply:
I think adding the feature that this function can be called from other threads than the master is outside the scope of this PR... As noted by George, there are accesses that are unsafe in this function, and we should keep the assert for now.